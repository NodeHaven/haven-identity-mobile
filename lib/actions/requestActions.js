// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  HANDLE_URL,
  HANDLE_MESSAGE,
  SELECT_REQUEST,
  SELECT_ACCOUNT_FOR_REQUEST,
  SAVE_REQUEST,
  CLEAR_REQUEST,
  AUTHORIZE_REQUEST,
  CANCEL_REQUEST,
  CALLBACK,
  AUTHORIZE_ACCOUNT
} from 'uPortMobile/lib/constants/RequestActionTypes'

export function handleURL (url, options = {}) {
  return {
    type: HANDLE_URL,
    url,
    ...{postback: false, popup: true, clientId: null},
    ...options
  }
}

export function handleMessage (message, options = {}) {
  return {
    type: HANDLE_MESSAGE,
    message,
    ...options
  }
}

export function selectRequest (activityId) {
  return {
    type: SELECT_REQUEST,
    activityId
  }
}

// Use this in a request dialog to select the correct account
export function selectAccountForRequest (activityId, account) {
  return {
    type: SELECT_ACCOUNT_FOR_REQUEST,
    activityId,
    account
  }
}

export function saveRequest (activityId) {
  return {
    type: SAVE_REQUEST,
    activityId
  }
}

export function clearRequest () {
  return {
    type: CLEAR_REQUEST
  }
}

export function authorizeRequest (activityId) {
  return {
    type: AUTHORIZE_REQUEST,
    activityId
  }
}

export function authorizeAccount (activityId, accType) {
  return {
    type: AUTHORIZE_ACCOUNT,
    activityId: activityId,
    accType: accType
  }
}

export function cancelRequest (activityId) {
  return {
    type: CANCEL_REQUEST,
    activityId
  }
}

export function callback (request) {
  return {
    type: CALLBACK,
    request
  }
}
