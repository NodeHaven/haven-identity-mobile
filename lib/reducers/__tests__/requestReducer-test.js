// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//

import reducer from 'uPortMobile/lib/reducers/requestReducer.js'
import { saveRequest, clearRequest } from 'uPortMobile/lib/actions/requestActions.js'

it('sets request id', () => {
  expect(reducer(null, saveRequest(123))).toMatchSnapshot()
})

it('replaces request id', () => {
  expect(reducer(124, saveRequest(123))).toMatchSnapshot()
})

it('clears request id', () => {
  expect(reducer(124, clearRequest())).toMatchSnapshot()
})

it('leaves a null request id as is', () => {
  expect(reducer(null, clearRequest())).toMatchSnapshot()
})

it('ignores unsupported action', () => {
  expect(reducer(undefined, {type: 'UNSUPPORTED'})).toEqual(null)
  expect(reducer(null, {type: 'UNSUPPORTED'})).toEqual(null)
  expect(reducer(124, {type: 'UNSUPPORTED'})).toEqual(124)
})
