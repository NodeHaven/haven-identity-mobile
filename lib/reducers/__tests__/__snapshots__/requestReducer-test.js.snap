// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clears request id 1`] = `null`;

exports[`leaves a null request id as is 1`] = `null`;

exports[`replaces request id 1`] = `123`;

exports[`sets request id 1`] = `123`;
